pkg/ccr/base/spec.go

```diff
@@ -1481,10 +1466,32 @@ func (s *Spec) RenameRollup(destTableName, oldRollup, newRollup string) error {
 func (s *Spec) DropRollup(destTableName, rollup string) error {
 	dbName := utils.FormatKeywordName(s.Database)
 	destTableName = utils.FormatKeywordName(destTableName)
-	rollup = utils.FormatKeywordName(rollup)
-	dropRollupSql := fmt.Sprintf("ALTER TABLE %s.%s DROP ROLLUP %s", dbName, destTableName, rollup)
-	log.Infof("drop rollup sql: %s", dropRollupSql)
-	return s.Exec(dropRollupSql)
+	descTableRes, err := s.queryResult(fmt.Sprintf("DESC %s.%s ALL", dbName, destTableName), "IndexName", "DESC TABLE")
+	if err != nil {
+		return xerror.Wrap(err, xerror.Normal, fmt.Sprintf("desc table %s failed", destTableName))
+	}
+	sort.Strings(descTableRes)
+
+	if index := sort.SearchStrings(descTableRes, rollup); index < len(descTableRes) && descTableRes[index] == rollup {
+		rollup = utils.FormatKeywordName(rollup)
+		dropRollupSql := fmt.Sprintf("ALTER TABLE %s.%s DROP ROLLUP %s", dbName, destTableName, rollup)
+		log.Infof("drop rollup sql: %s", dropRollupSql)
+		return s.Exec(dropRollupSql)
+	}
+	log.Info(fmt.Sprintf("rollup %s not found in %s, skip drop rollup", rollup, destTableName))
+	return nil
+}
+
+func (s *Spec) CheckRollupIndexExists(destTableName, rollup string) (bool, error) {
+	descTableRes, err := s.queryResult(fmt.Sprintf("DESC %s.%s ALL", s.Database, destTableName), "IndexName", "DESC TABLE")
+	if err != nil {
+		return false, xerror.Wrap(err, xerror.Normal, fmt.Sprintf("desc table %s failed", destTableName))
+	}
+	sort.Strings(descTableRes)
+	if index := sort.SearchStrings(descTableRes, rollup); index < len(descTableRes) && descTableRes[index] == rollup {
```

Comment:
`slice.Contains` is enough

Reply:
It seems that the `slice` pkg was added in go 1.24. You would use `utils.Contains` which is defined in https://github.com/selectdb/ccr-syncer/pull/416/.